File: activity/client/src/game/BonusType.ts
export enum BonusType {
  BOOST_CONSONNES  = "BOOST_CONSONNES",
  BOOST_VOYELLES   = "BOOST_VOYELLES",
  BOOST_FEU        = "BOOST_FEU",
  BOOST_EAU        = "BOOST_EAU",
  BOOST_TERRE      = "BOOST_TERRE",
  BOOST_VENT       = "BOOST_VENT",
  BOOST_RARES      = "BOOST_RARES",
  BOOST_ATK_GLOBAL = "BOOST_ATK_GLOBAL",
}

interface BonusDef { cibleType: string; statCiblee: string; mult: number }

const DEFS: Record<BonusType, BonusDef> = {
  [BonusType.BOOST_CONSONNES]:  { cibleType: "CONSONNES", statCiblee: "atk",           mult: 1.5 },
  [BonusType.BOOST_VOYELLES]:   { cibleType: "VOYELLES",  statCiblee: "atkspeed",       mult: 1.5 },
  [BonusType.BOOST_FEU]:        { cibleType: "FEU",       statCiblee: "burnDuration",   mult: 2.0 },
  [BonusType.BOOST_EAU]:        { cibleType: "EAU",       statCiblee: "slowAmount",     mult: 2.0 },
  [BonusType.BOOST_TERRE]:      { cibleType: "TERRE",     statCiblee: "armorReduction", mult: 1.2 },
  [BonusType.BOOST_VENT]:       { cibleType: "VENT",      statCiblee: "recoilCases",    mult: 2.0 },
  [BonusType.BOOST_RARES]:      { cibleType: "RARES",     statCiblee: "*",              mult: 1.3 },
  [BonusType.BOOST_ATK_GLOBAL]: { cibleType: "ALL",       statCiblee: "atk",            mult: 1.2 },
};

export function getBonusDef(b: BonusType): BonusDef { return DEFS[b]; }
export const ALL_BONUS_TYPES = Object.values(BonusType);

File: src/BonusType.java
public enum BonusType {
    BOOST_CONSONNES("CONSONNES", "atk",            1.5),
    BOOST_VOYELLES ("VOYELLES",  "atkspeed",        1.5),
    BOOST_FEU      ("FEU",       "burnDuration",    2.0),
    BOOST_EAU      ("EAU",       "slowAmount",      2.0),
    BOOST_TERRE    ("TERRE",     "armorReduction",  1.2),  // 1.25 * 1.2 = 1.5 effective
    BOOST_VENT     ("VENT",      "recoilCases",     2.0),
    BOOST_RARES    ("RARES",     "*",               1.3),  // "*" = toutes stats
    BOOST_ATK_GLOBAL("ALL",      "atk",             1.2);

    private final String cibleType;
    private final String statCiblee;
    private final double multiplicateur;

    BonusType(String cibleType, String statCiblee, double multiplicateur) {
        this.cibleType = cibleType;
        this.statCiblee = statCiblee;
        this.multiplicateur = multiplicateur;
    }

    public String getCibleType()     { return cibleType; }
    public String getStatCiblee()    { return statCiblee; }
    public double getMultiplicateur(){ return multiplicateur; }
}
